hypothesis-python/src/hypothesis/internal/conjecture/data.py

```diff
@@ -645,16 +640,16 @@ def start_example(self, i: int, label_index: int) -> None:
 
     class _mutator_groups(ExampleProperty):
         def begin(self) -> None:
-            self.groups: "Dict[Tuple[int, int], List[int]]" = defaultdict(list)
+            self.groups: "Dict[Tuple[int, int], List[int]]" = defaultdict(set)
 
         def start_example(self, i: int, label_index: int) -> None:
-            depth = len(self.example_stack)
-            self.groups[label_index, depth].append(i)
+            key = (self.examples[i].ir_start, self.examples[i].ir_end)
+            self.groups[label_index].add(key)
```

Comment:
`LazyStrategy` causes duplicate example boundaries, since best I can tell we wrap all of our core strategies in a lazy strategy. I've worked around that here by making groups unique up to `(ir_start, ir_end)`. The ideal is "don't create examples for LazyStrategy", but a naive fix there breaks `data.draw(label=...)` overrides, since they don't get passed to the underlying strategy call. May be doable with more plumbing.